### PR TITLE
Made Better Collada Exporter flake8-compliant

### PR DESCRIPTION
**Pull request by ghost**
The only thing I did not fix is the unused local variables because perhaps they were there for some reason, but those can be removed as well if that's not a concern.

Next changes should probably be:
- Use os.path.join instead of `... + "/" + ...`
- Use string formatting instead of string concatenation
- Use xml.etree.ElementTree instead of manually outputting XML 

I release all the changes in this PR under both the MIT license and the GPLv2 or later.

### CODE REVIEW

File: tools/export/blender25/godot_export_manager.py [hunk 1]
```diff
@@ -3,38 +3,42 @@
 # This program is free software; you can redistribute it and/or
 # modify it under the terms of the GNU General Public License
 # as published by the Free Software Foundation; either version 2
-# of the License, or (at your option) any later version.
+# of the License,  or (at your option) any later version.
```
Comment (akien-mga) [inline tools/export/blender25/godot_export_manager.py]: That's a bogus change :)

File: tools/export/blender25/godot_export_manager.py [hunk 2]
```diff
@@ -3,38 +3,42 @@
...
+import os
+from bpy.app.handlers import persistent
+from mathutils import Matrix
+
 bl_info = {
     "name": "Godot Export Manager",
     "author": "Andreas Esau",
-    "version": (1, 0),
-    "blender": (2, 7, 0),
+    "version": (1,  0),
```
Comment (akien-mga) [inline tools/export/blender25/godot_export_manager.py]: Bad change too, just running `sed -e 's/,/, /g'` is a bit too drastic if you don't check the diff afterwards...

**Comment by akien-mga** [comment]
I already noticed a couple issues at first glance. It would be nice if you could split the changes into separate commits. Cosmetic changes like adding spaces after commas should be done in their own commit, not together with using `xml.etree.ElementTree` or other functional changes.

**Comment by paper-pauper** [comment]
@akien-mga But that's what I did. Did you take a look at the diff?

File: tools/export/blender25/godot_export_manager.py [hunk 3]
```diff
@@ -44,8 +48,8 @@ class godot_export_manager(bpy.types.Panel):
 
     bpy.types.Scene.godot_export_on_save = BoolProperty(default=False)
 
-    ### draw function for all ui elements
-    def draw(self, context):
+    def draw(self,  context):
+        """ Draw function for all ui elements """
```
Comment (akien-mga) [inline tools/export/blender25/godot_export_manager.py]: This is not a cosmetic change, it's added documentation.

**Comment by akien-mga** [comment]
> But that's what I did: cosmetic changes only. Did you take a look at the diff?

Well that's my point. The diff is huge, so each commit should do *one* action, so that it can be only partly proofread and the rest can be assumed good. So there should be one commit to fix indentation, one commit to fix missing spaces after commas (with making it bogus double spaces after commas), and maybe a third commit for the other changes (no need to split *everything* into single commits, but every change that will impact 1000 lines should have its own commit).

**Comment by paper-pauper** [comment]
@akien-mga The changes took a lot of manual work; if I had split them into separate commits, I'd have to do it from scratch. So please accept my apologies for the bad approach but I suppose you could read it into chunks. In any case there are no functional changes, that's for sure.

**Comment by akien-mga** [comment]
Alright it's looking better now, so I guess splitting it in several commits is not forcefully needed. Could you fix the license header and squash the commits together?

File: tools/export/blender25/godot_export_manager.py [hunk 4]
```diff
@@ -199,122 +218,128 @@ class select_group_objects(bpy.types.Operator):
...
 
-        ### if path exists and group export name is set the group will be exported
-        if os.path.exists(path) and  group[self.idx].export_name != "":
-
-            context.scene.layers = [True,True,True,True,True,True,True,True,True,True,True,True,True,True,True,True,True,True,True,True]
+        # If path exists and group export name is set the group will be
+        # exported
+        if os.path.exists(path) and group[self.idx].export_name != "":
 
+            context.scene.layers = [True] * 20
```
Comment (akien-mga) [inline tools/export/blender25/godot_export_manager.py]: Nice simplification :D

File: tools/export/blender25/godot_export_manager.py [hunk 5]
```diff
@@ -323,17 +348,19 @@ def execute(self,context):
...
-            ### make particle duplicates, parent and select them
+            # Make particle duplicates, parent and select them
             nodes_to_be_added = []
             if group[self.idx].use_include_particle_duplicates:
-                for i,object in enumerate(group[self.idx].nodes):
+                for i, object in enumerate(group[self.idx].nodes):
                     if bpy.data.objects[object.name].type != "EMPTY":
-                        context.scene.objects.active = bpy.data.objects[object.name]
+                        context.scene.objects.active = bpy.data.objects[
+                            object.name]
```
Comment (akien-mga) [inline tools/export/blender25/godot_export_manager.py]: What's the limit for cropping lines in the tool you used? I'm not sure this change really adds to readability.

File: tools/export/blender25/godot_export_manager.py [hunk 6]
```diff
@@ -323,17 +348,19 @@ def execute(self,context):
...
                         bpy.data.objects[object.name].select = True
                         bpy.ops.object.duplicates_make_real()
                         for object in context.selected_objects:
                             nodes_to_be_added.append(object)
-                        bpy.ops.object.parent_set(type="OBJECT", keep_transform=False)
+                        bpy.ops.object.parent_set(type="OBJECT",
+                                                  keep_transform=False)
```
Comment (akien-mga) [inline tools/export/blender25/godot_export_manager.py]: Same here; does PEP8 really enforce cropping at 78-80 chars mechanically, or is it the tool you used that is a bit too drastic? In such cases I think common sense would be to not to crop if keeping a full statement on a single line means being between 80-85 chars. I'm no python guru, but that's the rule I apply visually in my own scripts :)

File: tools/export/blender25/io_scene_dae/__init__.py [hunk 7]
```diff
@@ -18,35 +18,30 @@
 
 # <pep8-80 compliant>
```
Comment (akien-mga) [inline tools/export/blender25/io_scene_dae/__init__.py]: That was a lie :P

File: tools/export/blender25/io_scene_dae/__init__.py [hunk 8]
```diff
@@ -70,29 +64,31 @@ class ExportDAE(bpy.types.Operator, ExportHelper):
...
             )
     use_mesh_modifiers = BoolProperty(
             name="Apply Modifiers",
             description="Apply modifiers to mesh objects (on a copy!).",
-	    default=False,
+            default=False,
             )
     use_tangent_arrays = BoolProperty(
-	    name="Tangent Arrays",
-	    description="Export Tangent and Binormal arrays (for normalmapping).",
```
Comment (akien-mga) [inline tools/export/blender25/io_scene_dae/__init__.py]: I wonder why this line gets wrapped, it's only 79 chars long.

**Comment by akien-mga** [comment]
> The changes took a lot of manual work; if I had split them into separate commits, I'd have to do it from scratch.

I don't know the tool you used, but if it doesn't allow to perform only selected changes using commandl line switches, it's worth an enhancement request upstream :)

**Comment by akien-mga** [comment]
Hm, I still think splitting at least the indentation change into a first commit before doing the other fixes would be welcome. The diff for `tools/export/blender25/io_scene_dae/export_dae.py` (not visible on GitHub since too big, but check the [raw patch](https://patch-diff.githubusercontent.com/raw/godotengine/godot/pull/5480.patch)) is unreadable due to indentation change from tabs to spaces in basically all the lines.

**Comment by akien-mga** [comment]
> Could you fix the license header and squash the commits together?

Ping :)

**Comment by paper-pauper** [comment]
@akien-mga Apologies, I'm having a busy week. I'll see if I can get it done next weekend, if that's not a problem.

**Comment by akien-mga** [comment]
Sure, there's no hurry :)

**Comment by paper-pauper** [comment]
@akien-mga Just a tip for the future: `git diff -w master blender-flake8` ignores all whitespace changes, so you can see the diff without them. Do you still want me to split the commit?

**Comment by akien-mga** [comment]
> Just a tip for the future: `git diff -w master blender-flake8` ignores all whitespace changes, so you can see the diff without them. Do you still want me to split the commit?

I know the `-w` switch, but many people don't, and git commits should be reviewable by anyone. It's a good practice to take in all case that cosmetic changes like a massive indentation change should be done in their own commit. So yeah if possible I'd prefer if you could split it.

> Edit: Also, the 79 character limit is required for PEP-8 compliance (see this)

I know too, though PEP-8 also says:
> Some teams strongly prefer a longer line length. For code maintained exclusively or primarily by a team that can reach agreement on this issue, it is okay to increase the nominal line length from 80 to 100 characters (effectively increasing the maximum length to 99 characters), provided that comments and docstrings are still wrapped at 72 characters.

I'm fine with using 80 chars, we don't need to forcefully go over, I just wondered what Python guys would do in such extreme cases like the ones I commented above, where the 79 char limit leads to overly cropped lines (which is bound to happen in a language where indentation is so important - i.e. being PEP-8 compliant means that it is not possible to have too many levels of indentation and still keep readable code...). I don't mean that you should change things, I just wanted to get your opinion on IMO sub-par changes that a forceful PEP-8 compliance induces :)

**Comment by paper-pauper** [comment]
@akien-mga The thing is that if the BCE is going to be maintained by the Blender folks as well, we should try to stick to neutral ground. The PEP-8 standard is well established, even if we might not like some things about it, and if we don't stick to it then either side can feel entitled to enforcing their own coding style.

The advantage is that BCE is going to pass tests under every static checker, even the strictest ones, without any special flags. I think that's good for ensuring that BCE's code stays clean in the future.

**Comment by akien-mga** [comment]
Yeah I agree following strict PEP-8 is likely the best in our case. I just mention that we have some corner cases where it really makes the code less readable - but no code style guidelines can be perfect.

**Comment by paper-pauper** [comment]
@akien-mga That's coding in a nutshell. If it were for me, I'd be using a language like Scala for everything, but for most things I'm forced to use Java or Python or JavaScript or whatever is needed....

That said, I split the changes into three commits, please give it a look!

**Comment by akien-mga** [comment]
Awesome, thanks a lot and sorry for being a bit perfectionist about diff readability ;)

**Pull request merged**